deploy-hub-configs/deploy.sh

```diff
@@ -34,14 +34,113 @@ if ./verify.sh; then
     sed -i "s/CHANGE_SPOKE_CLUSTERIMAGESET/${CLUSTERIMAGESET}/g" 02-cluster_imageset.yml
     sed -i "s%TAG_OCP_IMAGE_RELEASE%${NEWTAG}%g" 02-cluster_imageset.yml
 
+    # HACK for SNO to have DNS with DNSMasq
+    # We need clustername and IP's and create entries
+
+    echo ">> Detecting number of masters"
+    export NUM_M=$(oc --kubeconfig=${KUBECONFIG_HUB} get pods -A | grep coredns | wc -l)
```

Comment:
Should we also check if dnsmasq is already running in the hub?

Reply:
This is at HUB installation configuration, so this is the workaround in the case of SNO (where CoreDNS might not be there). How do suggest we should check that no Dnsmasq is there?

Reply:
>  How do suggest we should check that no Dnsmasq is there?

I was hoping you would figure this out :smile: 

> This is at HUB installation configuration, so this is the workaround in the case of SNO (where CoreDNS might not be there)

By default, CoreDNS *should not* be there. It is there because `kcli` adds it. If a HUB is installed using Assisted Installer's SaaS, it will have `dnsmasq` in it, not CoreDNS. Applying these configs blindly will overwrite the dnsmasq that AI installed.

Reply:
Now the check tries to reach MYIP port 53 and skip if needed